pkg/entities/entities_api_.go

```diff
@@ -0,0 +1,164 @@
+	query string,
+	sortBy []EntitySearchSortCriteria,
+) (*EntitySearch, error) {
+	return a.GetEntitySearchByQueryWithContext(context.Background(),
+		options,
+		query,
+		sortBy,
+	)
+}
+
+// Search for entities using a custom query.
+//
+// For more details on how to create a custom query
+// and what entity data you can request, visit our
+// [entity docs](https://docs.newrelic.com/docs/apis/graphql-api/tutorials/use-new-relic-graphql-api-query-entities).
+//
+// Note: you must supply either a `query` OR a `queryBuilder` argument, not both.
+func (a *Entities) GetEntitySearchByQueryWithContext(
+	ctx context.Context,
+	options EntitySearchOptions,
+	query string,
+	sortBy []EntitySearchSortCriteria,
+) (*EntitySearch, error) {
+
+	resp := entitySearchResponse{}
+	vars := map[string]interface{}{
+		"options": options,
+		"query":   query,
+		"sortBy":  sortBy,
+	}
+
+	if err := a.client.NerdGraphQueryWithContext(ctx, getEntitySearchByQuery, vars, &resp); err != nil {
+		return nil, err
+	}
+
+	return &resp.Actor.EntitySearch, nil
+}
+
+const getEntitySearchByQuery = `query(
+	$query: String,
+) { actor { entitySearch(
+	query: $query,
+) {
+	count
+	query
+	results {
+		entities {
+			__typename
+			accountId
+			alertSeverity
+			domain
+			entityType
+			guid
+			indexedAt
+			name
+			permalink
+			reporting
+			tags {
+				key
+				values
+			}
+			type
+			... on ApmApplicationEntityOutline {
+				__typename
+				applicationId
+				language
+			}
+			... on ApmDatabaseInstanceEntityOutline {
+				__typename
+				host
+				portOrPath
+				vendor
+			}
+			... on ApmExternalServiceEntityOutline {
+				__typename
+				host
+			}
+			... on BrowserApplicationEntityOutline {
+				__typename
+				agentInstallType
+				applicationId
+				servingApmApplicationId
+			}
+			... on DashboardEntityOutline {
+				__typename
+				createdAt
+				dashboardParentGuid
+				permissions
+				updatedAt
+			}
+			... on ExternalEntityOutline {
+				__typename
+			}
+			... on GenericEntityOutline {
+				__typename
+				tags {
+					key
+					values
+				}
+			}
+			... on GenericInfrastructureEntityOutline {
+				__typename
+				integrationTypeCode
+			}
+			... on InfrastructureAwsLambdaFunctionEntityOutline {
+				__typename
+				integrationTypeCode
+				runtime
+			}
+			... on InfrastructureHostEntityOutline {
+				__typename
+			}
+			... on MobileApplicationEntityOutline {
+				__typename
+				applicationId
+			}
+			... on SecureCredentialEntityOutline {
+				__typename
+				description
+				secureCredentialId
+				updatedAt
+			}
+			... on SyntheticMonitorEntityOutline {
+				__typename
+				monitorId
+				monitorType
+				monitoredUrl
+				period
+			}
+			... on ThirdPartyServiceEntityOutline {
+				__typename
+			}
+			... on UnavailableEntityOutline {
+				__typename
+			}
+			... on WorkloadEntityOutline {
+				__typename
+				createdAt
+				updatedAt
+			}
+		}
+		nextCursor
+	}
+	types {
+		count
+		domain
+		entityType
+		type
+	}
+} } }`
```

Comment:
I think it might also be useful to have a note added to `entities_api_.go` and `types_.go` which states something along the lines of "this code is NOT controlled by Tutone"

Reply:
Pretty sure this query supports tags for all types with the `tags` field at the top level of the response fields (lines 72-75).

Reply:
I get it now - this should be fine then :) except for one final find - the field `type` seems to have been excluded in this query.

https://www.diffchecker.com/2EpPBVXv/

Also, I apologize for the silly question, but just trying to be 100% sure - can we be sure that the addition of `tags` in the list of attribute common to all EntityOutline types would suffice - and we wouldn't need to add them per type?

Reply:
Yes, the `tags` field does get applied to the `ApmApplicationEntityOutline` type in the given GraphQL query. In GraphQL, when you query for a field that is part of an interface or a union, the fields within the fragment of a specific type will include the fields from the parent type or interface unless explicitly overridden.

Also the `type` field is on line 76 😉  It just moved during the code generation (my hunch is the iteration that's involved in codegen does not guarantee the order of the fields)

Reply:
how did I not see this - I'm so very sorry about that. I think we should be good with resolving this comment - thank you so much for explaining :)